### PR TITLE
Follow all player renames in roomgames

### PR DESCRIPTION
**Pull request by sirDonovan**
Fixes various crashes caused by roomgame players changing names outside
of the room

### CODE REVIEW

**Comment by Zarel** [comment]
???

This should not cause a crash.

If a player changes name while outside of the room, they should just be considered a different user...

**Comment by Zarel** [comment]
The API was kind of intentionally set up that way...

**Comment by Zarel** [comment]
I suppose straight-up following all renames would be easier.

File: simulator.js [hunk 1]
```diff
@@ -274,21 +274,7 @@ class Battle {
...
-			if (!this.players[user]) {
-				this.players[user] = player;
-				player.userid = user.userid;
-				player.name = user.name;
-				delete this.players[oldid];
-				player.simSend('rename', user.name, user.avatar);
-			}
-		}
-		if (!player && user in this.players) {
+		if (!(oldid in this.players) && user.userid in this.players) {
```
Comment (Zarel) [inline simulator.js]: ummm, the `allowRenames` check is _really fucking important_

Comment (Zarel) [inline simulator.js]: nvm I see it elsewhere

**Comment by sirDonovan** [comment]
One specific crash is when a tournament player leaves the room and changes names mid-battle. It would be simpler to fix that issue alone, but I think following all renames is better for roomgames going forward.

File: users.js [hunk 2]
```diff
@@ -674,6 +674,9 @@ class User {
 			this.connections[i].send(initdata);
 		}
 		this.named = false;
+		for (let i in this.games) {
+			this.games[i].renamePlayer(this, oldid);
```
Comment (Zarel) [inline users.js]: I think `onRename` might be a better name for this function... is it already used elsewhere or something?

File: simulator.js [hunk 3]
```diff
@@ -274,21 +274,7 @@ class Battle {
 	}
 	onRename(user, oldid) {
```
Comment (Zarel) [inline simulator.js]: Oh I guess it already is. Why don't we get rid of the old `onRename` and consolidate it under one?

**Comment by sirDonovan** [comment]
Updated.

**Pull request merged**